core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -525,6 +525,51 @@ boolean isLengthInvalidOrZero() {
     return this.length < 1;
   }
 
+  /**
+   *
+   * @return get length in bytes (dealing with special characters)
+   */
+  int getLengthInBytes( String value ) {
+    int total = 0;
```

Comment:
Why looping through each character in the string? The length in bytes can be found with just `value.getBytes( "UTF-8" ).length`. I can't see why this needs to be like this.